### ISSUE TITLE
no-dupe-args error thrown for nested array destructuring

### ISSUE THREAD

**Issue by dlwalsh** [eslint/eslint#3867]
I get the following:

```
  8:12  error  Duplicate param 'undefined'  no-dupe-args
```

for the following code.

```js
import React, { Component, PropTypes } from 'react';
import { geo } from 'd3';

const path = geo.path();

export default class SVGNode extends Component {

    viewBox([
        [left, top],
        [right, bottom]
    ]) {

        return [
            left,
            top,
            right - left, // width
            bottom - top // height
        ].join(' ');

    }

    render() {

        const { data } = this.props;

        return (
            <svg viewBox={this.viewBox(path.bounds(data))}>
                {data.features.map((feat, i) => (
                    <path key={i} d={path(feat.geometry)} />
                ))}
            </svg>
        );

    }
}

SVGNode.propTypes = {
    data: PropTypes.object
};
```

The code functions exactly as I expect it to.

I'm using eslint 1.5.0 and eslint-plugin-react 3.4.2.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by dlwalsh** [comment]
My .eslintrc looks like this:
```js
{
    "rules": {
        "indent": [2, 4],
        "quotes": [2, "single"],
        "linebreak-style": [2, "unix"],
        "semi": [2, "always"],
        "react/display-name": 0,
        "react/jsx-boolean-value": 1,
        "react/jsx-curly-spacing": 1,
        "react/jsx-max-props-per-line": 0,
        "react/jsx-no-duplicate-props": 1,
        "react/jsx-no-undef": 1,
        "react/jsx-quotes": 1,
        "react/jsx-sort-prop-types": 1,
        "react/jsx-sort-props": 0,
        "react/jsx-uses-react": 1,
        "react/jsx-uses-vars": 1,
        "react/no-danger": 1,
        "react/no-did-mount-set-state": 1,
        "react/no-did-update-set-state": 1,
        "react/no-multi-comp": 1,
        "react/no-unknown-property": 1,
        "react/prop-types": 1,
        "react/react-in-jsx-scope": 1,
        "react/require-extension": 1,
        "react/self-closing-comp": 1,
        "react/sort-comp": 1,
        "react/wrap-multilines": 1
    },
    "env": {
        "es6": true,
        "browser": true
    },
    "extends": "eslint:recommended",
    "ecmaFeatures": {
        "jsx": true,
        "modules": true,
        "experimentalObjectRestSpread": true
    },
    "plugins": [
        "react"
    ]
}
```

**Comment by mysticatea** [comment]
Thank you for the report.
I can reproduce this in the online demo.
I will work on this.

![image](https://cloud.githubusercontent.com/assets/1937871/9979530/1eae9770-5fad-11e5-901f-d0d414052866.png)
